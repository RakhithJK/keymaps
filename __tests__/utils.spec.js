import {
  reverseObjectMapping,
  reverseModifierScancodeMapping,
  generateRegionMap,
} from '../utils';
import mockRegion1 from './__mocks__/mockRegion1.json';

describe('utils', () => {
  describe('reverseObjectMapping', () => {
    test('should reverse key/value entries in object', () => {
      const obj = {
        1: 'a',
        2: 'b',
        3: 'c',
      };

      const expected = {
        a: '1',
        b: '2',
        c: '3',
      };
      expect(reverseObjectMapping(obj)).toEqual(expected);
    });
  });

  describe('reverseModifierScancodeMapping', () => {
    test('should reverse key/value entries in object, splitting them by their modifier', () => {
      const MODIFIER = '02';

      const obj = {
        '04': ['a', 'A'],
        '05': ['b', 'B'],
        '06': ['c', 'C'],
      };

      const expected = {
        a: '00,04',
        A: '02,04',
        b: '00,05',
        B: '02,05',
        c: '00,06',
        C: '02,06',
      };
      expect(reverseModifierScancodeMapping(obj, MODIFIER)).toEqual(
        expected,
      );
    });
  });

  describe('generateRegionMap', () => {
    test('should generate region map object', () => {
      expect(generateRegionMap(mockRegion1)).toMatchSnapshot();
    });

    test('should generate modified region map object', () => {
      expect(
        generateRegionMap(mockRegion1, (obj) => ({
          ...obj,
          extra: 'extra',
        })),
      ).toMatchSnapshot();
    });
  });
});
