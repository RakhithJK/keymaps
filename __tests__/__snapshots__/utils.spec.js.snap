// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils generateRegionMap should generate modified region map object 1`] = `
Object {
  "mockedRegion1": Object {
    "keys": Object {
      "04": Array [
        "a",
        "A",
      ],
      "extra": "extra",
    },
    "language": "mockedRegion1",
  },
}
`;

exports[`utils generateRegionMap should generate region map object 1`] = `
Object {
  "mockedRegion1": Object {
    "keys": Object {
      "04": Array [
        "a",
        "A",
      ],
    },
    "language": "mockedRegion1",
  },
}
`;
